yoti_python_sdk/client.py

```diff
@@ -69,31 +74,64 @@ def get_activity_details(self, encrypted_request_token):
         attribute_list = protobuf.Protobuf().attribute_list(decrypted_data)
         return ActivityDetails(receipt, attribute_list)
 
-    def __make_request(self, encrypted_request_token):
-        path = self.__get_request_path(encrypted_request_token)
+    def perform_aml_check(self, aml_profile):
+        if aml_profile is None:
+            raise TypeError("aml_profile not set")
+
+        http_method = 'POST'
+
+        response = self.__make_aml_check_request(http_method, aml_profile)
+
+        return aml.AmlResult(response.text)
+
+    def __make_activity_details_request(self, encrypted_request_token, http_method, content):
+        decrypted_token = self.__crypto.decrypt_token(encrypted_request_token).decode('utf-8')
+        path = self.__endpoint.get_activity_details_request_path(decrypted_token)
         url = yoti_python_sdk.YOTI_API_ENDPOINT + path
-        headers = self.__get_request_headers(path)
+        headers = self.__get_request_headers(path, http_method, content)
         response = requests.get(url=url, headers=headers)
 
         if not response.status_code == 200:
             raise RuntimeError('Unsuccessful Yoti API call: {0}'.format(response.text))
 
         return response
 
-    def __get_request_path(self, encrypted_request_token):
-        token = self.__crypto.decrypt_token(encrypted_request_token).decode('utf-8')
-        nonce = uuid.uuid4()
-        timestamp = int(time.time() * 1000)
+    def __make_aml_check_request(self, http_method, aml_profile):
+        aml_profile_json = json.dumps(aml_profile.__dict__)
+        aml_profile_bytes = aml_profile_json.encode()
+        path = self.__endpoint.get_aml_request_url()
+        url = yoti_python_sdk.YOTI_API_ENDPOINT + path
+        headers = self.__get_request_headers(path, http_method, aml_profile_bytes)
 
-        return '/profile/{0}?nonce={1}&timestamp={2}&appId={3}'.format(
-            token, nonce, timestamp, self.sdk_id
-        )
+        response = requests.post(url=url, headers=headers, data=aml_profile_bytes)
```

Comment:
This will become a bit repetitive if we end up having a lot of calls, but it's fine for now.